src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -782,4 +794,197 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+        return truncated_list, truncated_packages_removed, capacity - self.__get_byte_size(truncated_list)
+
+    def __read_complete_status_file_raw_data(self, file_path):
+        for i in range(0, Constants.MAX_FILE_OPERATION_RETRY_COUNT):
+            try:
+                with self.env_layer.file_system.open(file_path, 'r') as file_handle:
+                    status_file_data_raw = json.load(file_handle)[0]  # structure is array of 1
+            except Exception as error:
+                if i < Constants.MAX_FILE_OPERATION_RETRY_COUNT - 1:
+                    time.sleep(i + 1)
+                else:
+                    self.composite_logger.log_error(
+                        "Unable to read status file (retries exhausted). Error: {0}.".format(repr(error)))
+                    raise
+        return status_file_data_raw
+
+    def __get_latest_complete_status_file_path(self, status_folder_path):
+        """ Get the latest status complete file and remove other .complete.status files """
+        list_of_files = glob.glob(status_folder_path + '\\' + '*.complete.status')
+        latest_file = max(list_of_files, key=lambda x: (os.path.getmtime(x), int(re.search(r'(\d+)\.complete.status', x).group(1)), x))
+        self.composite_logger.log_debug("list of complete status files: {0}".format(list_of_files))
+
+        # Remove older complete.status files
+        for file in list_of_files:
+            if file != latest_file:
+                self.env_layer.file_system.delete_files_from_dir(file, '*.complete.status')
+
+        return os.path.realpath(latest_file)
+
+    def __set_truncated_package_detail(self, name, truncated_package_list):
+        """ Compose truncated packages for logging """
+        return {
+            "name": str.capitalize(name),
+            "truncated_packages": truncated_package_list
+        }
+
+    def __get_byte_size(self, val):
+        """ Get the current byte size of val """
+        return len(json.dumps(val).encode("utf-8"))
+
+    def __get_quote_count(self, package_list):
+        return sum(char == '"' for char in json.dumps(package_list)) if not len(package_list) == 0 else 0
+
+    def __recompose_truncated_status_payload(self, truncated_status_file, new_patches, code, errors_detail_list):
+        """ Recompose complete status file payload with new errors detail list, new errors message, new truncated patches, and Warning status  """
+        error_message = Constants.StatusTruncationConfig.TRUNCATION_ERROR_MESSAGE
+        truncated_error_detail = self.__set_error_detail(Constants.PatchOperationErrorCodes.TRUNCATION, error_message)
+        errors_detail_list.insert(0, truncated_error_detail)
+
+        # Max length of error details is set to 5
+        if len(errors_detail_list) > Constants.STATUS_ERROR_LIMIT:
+            errors_detail_list = errors_detail_list[:Constants.STATUS_ERROR_LIMIT]
+
+        truncated_errors_json = self.__set_truncation_errors_json(code, errors_detail_list)
```

Comment:
There is an existing logic to add errors in error objects. All errors should go in the list via that logic. Please re-use that